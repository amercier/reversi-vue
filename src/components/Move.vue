<template>
  <a
    href="#"
    :class="`move move--${color}`"
    v-on:click.prevent="$emit('clicked')"
  ></a>
</template>

<script>
import { COLORS } from '../config';

/**
 * Potential move button.
 */
export default {
  name: 'Move',
  props: {
    /**
     * Color of the player to play the move.
     *
     * @type {String}
     */
    color: {
      type: String,
      validator: value => COLORS.indexOf(value) !== -1,
      required: true,
    },
  },
};
</script>

<style lang="scss">
$colors: (
  white: (
    rgba(180, 180, 180, 0.8),
    rgba(255, 255, 255, 0.8),
  ),
  black: (
    #000,
    #161616,
  ),
);

.move {
  position: relative;

  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    border-radius: 50%;
    opacity: 0.4;
    transform: scale(0.25);
    transition: opacity 150ms ease-out, transform 150ms ease-out;
  }

  @each $color, $gradient in $colors {
    &--#{$color}::before {
      background: linear-gradient(
        to bottom,
        nth($gradient, 1),
        nth($gradient, 2)
      );
    }
  }

  &:hover::before,
  &:focus::before {
    opacity: 0.5;
    transform: scale(0.5);
  }
}
</style>
