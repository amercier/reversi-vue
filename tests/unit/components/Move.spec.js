import { shallowMount } from '@vue/test-utils';
import Move from '@/components/Move.vue';

describe('Move.vue', () => {
  it('renders a white move', () => {
    const move = shallowMount(Move, {
      propsData: { color: 'white', onClick: () => {} },
    });
    expect(move.html()).toMatchSnapshot();
  });

  it('renders a black move', () => {
    const move = shallowMount(Move, {
      propsData: { color: 'black', onClick: () => {} },
    });
    expect(move.html()).toMatchSnapshot();
  });
});
