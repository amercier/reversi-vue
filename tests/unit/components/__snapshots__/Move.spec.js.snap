// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Move.vue renders a black move 1`] = `<a href="#" class="move move--black"></a>`;

exports[`Move.vue renders a white move 1`] = `<a href="#" class="move move--white"></a>`;
